qiita_pet/handlers/artifact_handlers/base_handlers.py

```diff
@@ -0,0 +1,342 @@
+                raise QiitaHTTPError(403, "Access denied to artifact %s"
+                                          % artifact.id)
+        elif analysis:
+            if not analysis.has_access(user):
+                raise QiitaHTTPError(403, "Access denied to artifact %s"
+                                          % artifact.id)
+        else:
+            # This can't happen but worth adding a check
+            raise QiitaHTTPError(500, "Error accessing artifact %s"
+                                      % artifact.id)
+
+
+def artifact_summary_get_request(user, artifact_id):
+    """Returns the information for the artifact summary page
+
+    Parameters
+    ----------
+    user : qiita_db.user.User
+        The user making the request
+    artifact_id : int or str
+        The artifact id
+
+    Returns
+    -------
+    dict of objects
+        A dictionary containing the artifact summary information
+        {'name': str,
+         'artifact_id': int,
+         'visibility': str,
+         'editable': bool,
+         'buttons': str,
+         'processing_parameters': dict of {str: object},
+         'files': list of (int, str),
+         'processing_jobs': list of [str, str, str, str, str],
+         'summary': str or None,
+         'job': [str, str, str],
+         'errored_jobs': list of [str, str]}
+    """
+    artifact_id = int(artifact_id)
+    artifact = Artifact(artifact_id)
+
+    check_artifact_access(user, artifact)
+
+    visibility = artifact.visibility
+    summary = artifact.html_summary_fp
+    job_info = None
+    errored_jobs = []
+    processing_jobs = []
+    for j in artifact.jobs():
+        if j.command.software.type == "artifact transformation":
+            status = j.status
+            if status == 'success':
+                continue
+            j_msg = j.log.msg if status == 'error' else None
+            processing_jobs.append(
+                [j.id, j.command.name, j.status, j.step, j_msg])
+
+    # Check if the HTML summary exists
+    if summary:
+        with open(summary[1]) as f:
+            summary = f.read()
+    else:
+        # Check if the summary is being generated
+        command = Command.get_html_generator(artifact.artifact_type)
+        all_jobs = set(artifact.jobs(cmd=command))
+        jobs = [j for j in all_jobs if j.status in ['queued', 'running']]
+        errored_jobs = [(j.id, j.log.msg)
+                        for j in all_jobs if j.status in ['error']]
+        if jobs:
+            # There is already a job generating the HTML. Also, there should be
+            # at most one job, because we are not allowing here to start more
+            # than one
+            job = jobs[0]
+            job_info = [job.id, job.status, job.step]
+
+    # Check if the artifact is editable by the given user
+    study = artifact.study
+    analysis = artifact.analysis
+    editable = study.can_edit(user) if study else analysis.can_edit(user)
+
+    buttons = []
```

Comment:
I agree - however this code was already there, I just adapted it from a different file. Changing this would require some more thought on the interface design so I would suggest just merge it and refactor later if needed.